code-challenges.test.js

```diff
@@ -15,42 +15,220 @@
+  })
+  return temp
+}
+
+// The test passed (green) :D
+
+
+
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
-// Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
-// Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
-// Expected output: false
+// a describe method that lists the name of the function OR naming of the particular test.
+describe("fullHouse", () => {
+
+  // a test/it method, nested within the describe block, that in plain words, describes that the function does.
+  it("takes in an array of 5 numbers and determines whether or not the array is a full house", () => {
+    const hand1 = [5, 5, 5, 3, 3]
+    // Expected output: true
+    const hand2 = [5, 5, 3, 3, 4]
+    // Expected output: false
+    const hand3 = [5, 5, 5, 5, 4]
+    // Expected output: false
+
+    //an expect method, nested within the test block, calling on the hello() function, followed by the .toEqual() matcher that checks the expected output of the function return.
+    expect(fullHouse(hand1)).toEqual(true)
+    expect(fullHouse(hand2)).toEqual(false)
+    expect(fullHouse(hand3)).toEqual(false)
+  })
+})
+
+// fullHouse › takes in an array of 5 numbers and determines whether or not the array is a full house
+// ReferenceError: fullHouse is not defined
+// Good failure
 
 
 // b) Create the function that makes the test pass.
+
+// I wasn't able to figure this one out unfortunately but I'll explain my thought process when tackling this challenge.
+// The core idea I had in mind was to set up three variables that will be assigned the boolean value of false within the function. The first variable, threeOfAKind, is checking to see if there is a three of a kind within the array and if that is the case, then I would reassign that value to true.
+// The second variable, pair, is also set to false. This variable is checking to see if a pair exists within the array and if that is the case, then I would reassign that value to true.
+// The last variable, feelingLucky, is set to false. This variable is checking to see that if both threeOfAKind AND pair are true, then feelingLucky will be reassigned to true, meaning that a full house exists in the inputted array.
+// This challenge was difficult and I ended up setting up two nested for-loops to check and see if a three of a kind and a pair exists. I also set up a count variable that would increment up everytime there was a duplicate value in the array. If that count variable hit a certain value such as 3, that would indicate that there three values that are the same within the array, hence a three of a kind. The same logic applies for the variable, pair.
+// Unfortunately, running this didn't pass the test.
```

Comment:
Love this idea! I think there is something here where you can create an object to manage a state and pop values from the array into another array and check it's length. Or check how many instances of a value pop up in the array by initially saving a value and every time  you find that value in the array incrementing a value in te object. Just some ideas